### PR TITLE
chore: fix typecheck + enable python 3.11

### PR DESCRIPTION
**Pull request by tas17**

### CODE REVIEW

File: saiph/tests/benchmark_test.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ def test_memory_iris(record_property: Any, iris_df: pd.DataFrame) -> None:
     fit(iris_df)
     peak = int(getrusage(RUSAGE_SELF).ru_maxrss / 1024)
     # memory usage should be below x kiB
-    error_margin = 1.1
+    error_margin = 1.2
```
Comment (tas17) [inline saiph/tests/benchmark_test.py:15]: On m1, about 1 out of 2 times the tests were failing because of that

File: pyproject.toml [hunk 2]
```diff
@@ -27,7 +28,7 @@ jupyter = "^1.0"
 isort = "^5.9"
 pytest = "^6.2"
 pydocstyle = "^6.1"
-mypy = "^0.920"
+mypy = "^0.991"
```
Comment (tcrasset) [inline pyproject.toml:31]: Can you do a `poetry show -lo` and update all the other packages, too?

File: saiph/reduction/famd_sparse.py [hunk 3]
```diff
@@ -18,6 +18,9 @@ def fit(
     df: pd.DataFrame,
     nf: Optional[int] = None,
     col_weights: Optional[NDArray[np.float_]] = None,
+    seed: Optional[
+        int
+    ] = None,  # Even if not used, a fit function accepts a seed argument (typecheck)
```
Comment (tcrasset) [inline saiph/reduction/famd_sparse.py:23]:
```suggestion
    ] = None,
```

This is good as a PR comment, but I don't think it's needed as a comment in the code.

File: saiph/tests/benchmark_test.py (hunk 1 above)
Comment (tcrasset) [inline saiph/tests/benchmark_test.py:15]: You should increase the memory required instead of the error margin.

**Pull request merged**